microsoft/PowerToys
# [FancyZones] Editor multi monitor support

## Summary of the Pull Request

Multi-monitor support.

## PR Checklist
* [x] Applies to #6435

## Info on Pull Request

Layout and layout settings could be selected for each monitor. Any change would be saved after clicking `Apply`. For now, the `Apply` button doesn't close the editor window.
On the active monitor selection, layout and settings are shown accordingly to the monitor (except switching `Templates` and `Custom` tabs)

![image](https://user-images.githubusercontent.com/8949536/92915884-a8c2bc80-f435-11ea-9f35-6093b9c65919.png)
![image](https://user-images.githubusercontent.com/8949536/92915916-b11af780-f435-11ea-8112-0d9c0b429bf6.png)
![image](https://user-images.githubusercontent.com/8949536/92916009-c8f27b80-f435-11ea-89d5-0bdef3dd7d5b.png)


Displays are presented with their relative dimensions.
![1](https://user-images.githubusercontent.com/8949536/93221873-ef3a5300-f776-11ea-8a57-8a3ac12f0987.PNG)
![2](https://user-images.githubusercontent.com/8949536/93221880-f06b8000-f776-11ea-813d-9d830a04a00c.PNG)
![3](https://user-images.githubusercontent.com/8949536/93221882-f19cad00-f776-11ea-8708-23592466d79b.PNG)

## Validation Steps Performed

Use the editor with different desktops configurations. 
Verify:
* settings were saved properly for each desktop. 
* zones are placed properly for each desktop
    * in the editor preview 
    * while editing
    * after being applied
* grid and canvas editors don't crash.
* template settings (`Zone count`, `Show space around zones`, `Space around zones`) are saved individually for each desktop and those values are changed on the desktop switch in the editor.
